thirdparty/s3.go

```diff
@@ -179,7 +189,187 @@ func PutS3File(pushAuth *aws.Auth, localFilePath, s3URL, contentType, permission
 		"problem putting %s to bucket", localFilePath)
 }
 
-func GetS3File(auth *aws.Auth, s3URL string) (io.ReadCloser, error) {
+func PutS3File(pushAuth *aws.Auth, localFilePath, s3URL, contentType, permissionACL string) error {
+	if !newCode {
+		return legacyPutS3File(pushAuth, localFilePath, s3URL, contentType, permissionACL)
+	}
+
+	urlParsed, err := url.Parse(s3URL)
+	if err != nil {
+		return err
+	}
+
+	if urlParsed.Scheme != "s3" {
+		return errors.Errorf("Don't know how to use URL with scheme %v", urlParsed.Scheme)
+	}
+
+	config := &awsSDK.Config{
+		Credentials: credentials.NewStaticCredentials(pushAuth.AccessKey, pushAuth.SecretKey, pushAuth.Token()),
+		Region:      awsSDK.String(region),
+	}
+	session, err := session.NewSession(config)
+	if err != nil {
+		return errors.Wrap(err, "error creating new session")
+	}
+	svc := awsS3.New(session)
+	bucket := awsS3.CreateBucketInput{
+		Bucket: &urlParsed.Host,
+	}
+
+	file, err := os.Open(localFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "Error opening file %s", localFilePath)
+	}
+	defer file.Close()
+
+	// Get size of file to read into buffer
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return errors.Wrapf(err, "Error getting stats for file %s", localFilePath)
+	}
+	size := fileInfo.Size()
+	buffer := make([]byte, size)
+	_, err = file.Read(buffer)
```

Comment:
Since you're reading the entire file, a simpler way to do this would be to use ioutil.ReadAll